beacon_chain/validation/src/attestation_parent_hashes.rs

```diff
@@ -0,0 +1,226 @@
+use super::types::Hash256;
+
+#[derive(Debug)]
+pub enum ParentHashesError {
+    BadCurrentHashes,
+    BadObliqueHashes,
+    SlotTooHigh,
+    SlotTooLow,
+    IntWrapping,
+}
+
+/// This function is used to select the hashes used in
+/// the signing of an AttestationRecord.
+///
+/// It either returns Result with a vector of length `cycle_length,` or
+/// returns an Error.
+///
+/// This function corresponds to the `get_signed_parent_hashes` function
+/// in the Python reference implentation.
+///
+/// See this slide for more information:
+/// https://tinyurl.com/ybzn2spw
+pub fn attestation_parent_hashes(
+    cycle_length: u8,
+    block_slot: u64,
+    attestation_slot: u64,
+    current_hashes: &[Hash256],
+    oblique_hashes: &[Hash256])
+    -> Result<Vec<Hash256>, ParentHashesError>
+{
+    // This cast places a limit on cycle_length. If you change it, check math
+    // for overflow.
+    let cycle_length: u64 = u64::from(cycle_length);
+
+    if current_hashes.len() as u64 != (cycle_length * 2) {
+        return Err(ParentHashesError::BadCurrentHashes);
+    }
+    if oblique_hashes.len() as u64 > cycle_length {
+        return Err(ParentHashesError::BadObliqueHashes);
+    }
+    if attestation_slot >= block_slot {
+        return Err(ParentHashesError::SlotTooHigh);
+    }
+
+    /*
+     * Cannot underflow as block_slot cannot be less
+     * than attestation_slot.
+     */
+    let attestation_distance = block_slot - attestation_slot;
+
+    if attestation_distance > cycle_length {
```

Comment:
Note that there is a bug in the current spec that makes this true. What we want to be able to do is include and process attestations relative to the parent block. Expect a change to spec regarding "recent_block_hashes" and how parent_hashes are calculated in the next day or so.

Nothing major, but will slightly change some of the subtle checks here

Reply:
This all looks fine, but I'm not doing extensive review of this function until we fix the spec

Reply:
Hey @djrtwo -- this is the last comment I need to address. Do you think I should update my code as per the spec now? The alternatives are (a) wait because more spec changes are coming, (b) implement the update in a new PR. Just looking for input :)

Reply:
After some more thought, I am leaning towards a new PR which updates this code to the latest spec. The changes involved are significant and would be easier to digest in a single PR, I think.